beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -118,6 +118,31 @@ func (vs *Server) PrepareBeaconProposer(
 	return &emptypb.Empty{}, nil
 }
 
+// GetFeeRecipientByPubKey returns a fee recipient from the beacon node's settings or db based on a given public key
+func (vs *Server) GetFeeRecipientByPubKey(ctx context.Context, request *ethpb.FeeRecipientByPubKeyRequest) (*ethpb.FeeRecipientByPubKeyResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "validator.GetFeeRecipientByPublicKey")
+	defer span.End()
+	if request == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "request was empty")
+	}
+
+	resp, err := vs.ValidatorIndex(ctx, &ethpb.ValidatorIndexRequest{PublicKey: request.PublicKey})
+	if err != nil {
+		return &ethpb.FeeRecipientByPubKeyResponse{
+			FeeRecipient: params.BeaconConfig().DefaultFeeRecipient.Bytes(),
+		}, nil
+	}
+	address, err := vs.BeaconDB.FeeRecipientByValidatorID(ctx, resp.GetIndex())
+	if err != nil {
+		return &ethpb.FeeRecipientByPubKeyResponse{
+			FeeRecipient: params.BeaconConfig().DefaultFeeRecipient.Bytes(),
+		}, nil
```

Comment:
This is rather dangerous, and we shouldn't skip out on the error and just default to burn address.
The caller should be able to handle the error with a switch statement

Reply:
> This is rather dangerous, and we shouldn't skip out on the error and just default to burn address.
> 
> The caller should be able to handle the error with a switch statemen
rotsen11

Reply:
I'll handle this better